### PR TITLE
Feature/ak-49: Added AWS KMS App-Tier CMK plugin, test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/kms/kmsAppTierCmk.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    category: 'KMS',
+    description: 'Ensures that there is one Amazon KMS Customer Master Key (CMK) present in the account for App-Tier resources.',
+    more_info: 'Amazon KMS should have Customer Master Key (CMK) for App-Tier to protect data in transit.',
+    recommended_action: 'Create a Customer Master Key (CMK) with App-Tier tag',
+    link: 'https://docs.aws.amazon.com/kms/latest/developerguide/create-keys.html',
+    apis: ['KMS:listKeys', 'ResourceGroupsTaggingAPI:getTagKeys', 'KMS:listResourceTags'],
+    settings: {
+        kms_cmk_tag_key: {
+            name: 'KMS CMK Tag Key',
+            description: 'When this tag key exists in the AWS account, there should also exist a KMS CMK having same tag key',
```
Comment (giorod3) [inline plugins/aws/kms/kmsAppTierCmk.js:15]:
```suggestion
            description: 'When this tag key exists in the AWS account, there should also exist a KMS CMK with the same tag key',
```

**Review by giorod3 — COMMENTED** [review]
small syntax change. Ill make the commit

**Review by giorod3 — APPROVED** [review]

File: collectors/aws/collector.js [hunk 2]
```diff
@@ -1257,6 +1269,7 @@ var collect = function(AWSConfig, settings, callback) {
         }, function() {
             callback(null, collection);
         });
+        // console.log(JSON.stringify(collection, null, 2));
```
Comment (dipsubha06) [inline collectors/aws/collector.js:1272]: remove commented code

File: plugins/aws/kms/kmsAppTierCmk.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            }
+
+            if (!getTagKeys.data.length) {
+                helpers.addResult(results, 0, 'No tag keys found', region);
+                return rcb();
+            }
+
+            if (!getTagKeys.data.includes(config.kms_cmk_tag_key)) {
+                helpers.addResult(results, 0,
+                    `No key with "${config.kms_cmk_tag_key}" tag found`, region);
```
Comment (dipsubha06) [inline plugins/aws/kms/kmsAppTierCmk.js:50]: Why are we returning 0 when no key witth kms_cmk_tag_key found as we are reuring 2 below for the same

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

File: collectors/aws/collector.js (hunk 2 above)
Comment (AkhtarAmir) [reply collectors/aws/collector.js]: Removed

File: plugins/aws/kms/kmsAppTierCmk.js (hunk 3 above)
Comment (AkhtarAmir) [reply plugins/aws/kms/kmsAppTierCmk.js:50]: Updated result code to 2

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**